scanner.go

```diff
@@ -55,9 +55,28 @@ func NewScanner(evaluator Evaluator) *Scanner {
 // errors and returning at the end of input.
 func (s *Scanner) Transform(input []byte) ([]byte, error) {
 	errs := &source.Errors{}
-	for _, char := range string(input) {
-		if err := s.consumeWithError(char); err != nil {
-			errs.Push(err)
+	lines := bytes.Split(input, []byte("\n"))
+
+	for ix, line := range lines {
+		// If the line starts with a comment, we don't want to evaluate it.
+		// We currently only support comments starting with a hash (#).
+		trimmedLine := bytes.TrimSpace(line)
```

Comment:
I think we should not trim? We probably want to keep supporting this no? 
![image](https://user-images.githubusercontent.com/40318863/223159549-3a03992b-9cf8-40c6-8355-ad8c6206abc0.png)

Reply:
I see, so lets not trim space I guess?

Reply:
Yes @peterdeme